contrib/zpages/src/main/java/io/opencensus/contrib/zpages/StatszZPageHandler.java

```diff
@@ -554,22 +559,31 @@ private StatszZPageHandler(ViewManager viewManager) {
    * directories, and leaf nodes are the ones with actual information on views. Each non-leaf node
    * also has the number of views under its directory.
    */
-  private static class TreeNode {
+  @AutoValue
+  // Suppress Checker Framework warning about missing @Nullable in generated equals method.
+  @AutoValue.CopyAnnotations
+  @SuppressWarnings("nullness")
+  abstract static class TreeNode {
```

Comment:
Is TreeNode immutable?  It looks like `children` isn't populated in the factory method.

Reply:
No it cannot be immutable, since we can register views after StatsZ is initialized, and that will update all the parent nodes of the newly registered view. (Specifically, root will be updated every time when there's a new view.)

Reply:
Hmmm in this case I shouldn't use AutoValue for TreeNode... I've reverted the last 2 commits.